contracts/interfaces/IDCAStrategies.sol

```diff
@@ -0,0 +1,235 @@
+    bytes32 s
+  ) external;
+
+  /**
+   * @notice Sets permissions via signature
+   * @dev This method works similarly to `modify`, but instead of being executed by the owner, it can be set my signature
+   * @param permissions The permissions to set
+   * @param tokenId The token's id
+   * @param deadline The deadline timestamp by which the call must be mined for the approve to work
+   * @param v Must produce valid secp256k1 signature from the holder along with `r` and `s`
+   * @param r Must produce valid secp256k1 signature from the holder along with `v` and `s`
+   * @param s Must produce valid secp256k1 signature from the holder along with `r` and `v`
+   */
+  function permissionPermit(
+    IDCAStrategies.PermissionSet[] calldata permissions,
+    uint256 tokenId,
+    uint256 deadline,
+    uint8 v,
+    bytes32 r,
+    bytes32 s
+  ) external;
+
+  /**
+   * @notice Sets a new NFT descriptor
+   * @dev Will revert with ZeroAddress if address is zero
+   * @param descriptor The new NFT descriptor contract
+   */
+  function setNFTDescriptor(IDCAHubPositionDescriptor descriptor) external;
+}
+
+interface IDCAStrategiesPositionsHandler {
+  function deposit(IDCAStrategies.DepositParams calldata parameters) external returns (uint256);
+
+  function withdrawSwapped(uint256 positionId, address recipient) external returns (uint256);
+
+  function increasePosition(
+    uint256 positionId,
+    uint256 amount,
+    uint32 newSwaps
+  ) external;
+
+  function reducePosition(
+    uint256 positionId,
+    uint256 amount,
+    uint32 newSwaps,
+    address recipient
+  ) external;
+
+  function terminate(
+    uint256 positionId,
+    address recipientUnswapped,
+    address recipientSwapped
+  ) external returns (uint256 unswapped, uint256 swapped);
+
+  function syncPositionToLatestStrategyVersion(uint256 positionId) external;
+
+  function increaseAndSyncPositionToLatestStrategyVersion(
+    uint256 positionId,
+    uint256 amount,
+    uint32 newSwaps
+  ) external;
+
+  function reduceAndSyncPositionToLatestStrategyVersion(
+    uint256 positionId,
+    uint256 amount,
+    uint32 newSwaps,
+    address recipient
+  ) external;
+}
+
+interface IDCAStrategies is IDCAStrategiesManagementHandler, IDCAStrategiesPermissionsHandler, IDCAStrategiesPositionsHandler {
+  enum Permission {
+    INCREASE,
+    REDUCE,
+    WITHDRAW,
+    TERMINATE,
+    SYNC
+  }
+
+  struct DepositParams {
+    IDCAHub hub;
+    uint80 strategyId;
+    address from;
+    uint256 amount;
+    uint32 amountOfSwaps;
+    uint32 swapInterval;
+    address owner;
+    PermissionSet[] permissions;
+  }
+
+  struct Position {
+    IDCAHub hub; // 20 bytes
+    uint80 strategyId; // 10 bytes
+    uint16 strategyVersion; // 2 bytes
+    uint256[] positions;
+  }
+
+  struct ShareOfToken {
+    address token;
+    uint80 share; // 0 < share < 100%
+  }
+
+  struct PermissionSet {
+    address operator;
+    Permission[] permissions;
+  }
+
+  struct Strategy {
```

Comment:
I think this one can be moved under `IDCAStrategiesManagementHandler`, I don't think it will be used by others